docs/DeveloperGuide.md

```diff
@@ -532,133 +535,213 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
-    * 2a1. gitGud shows an error message.
+    Use case ends.
 
-      Use case resumes at step 1.
+**Extensions**
 
+* 1a. gitGud detects that the `FRIEND_ID` does not exist.  
+    * 1a1. gitGud informs user that it has no friend was found with specified `FRIEND_ID`.  
+    * 1a2. User makes another request to delete a friend with a different `FRIEND_ID` for an existing friend.
+    * Use case continues from step 2.  
 
-**Use case: UC03 - List friends whose `FRIEND_ID` contains a keyword**
+**Use case: UC04 - Link a game to a friend**
 
 **MSS**
 
-1. User requests to list friends in gitGud using a keyword filter.
-2. gitGud shows a filtered list of friends whose `FRIEND_ID` contains the keyword.
+1. User requests to link a friend using a friend's `FRIEND_ID`, a game's `GAME_ID` and an `IN_GAME_USERNAME`.
+2. gitGud links the friend to the game, stores their in-game username and informs user of successful linking of friend.
 
    Use case ends.
 
 **Extensions**
 
-* 2a. The user does not provide a keyword.
+* 1a. gitGud detects that the `FRIEND_ID`/`GAME_ID` does not exist.
+    * 1a1. gitGud informs user that it has no friend/game was found with specified `FRIEND_ID`/`GAME_ID`.  
+    * 1a2. User makes another request to link a friend with a game, with a different `FRIEND_ID`/`GAME_ID` for an existing friend/game.  
+    * Use case continues from step 2.     
+
+* 1b. gitGud detects that the `IN_GAME_USERNAME` provided is invalid and does not meet specifications provided in User Guide.
+    * 1b1. gitGud informs user that `IN_GAME_USERNAME` provided is invalid.
+    * 1b2. User makes another request to link a friend with a game, with existing `FRIEND_ID` and `GAME_ID` and valid `IN_GAME_USERNAME`.
+    * Use case continues from step 2.
 
-    * 2a1. All friends are listed.
+**Use case: UC05 - Unlink a game from a friend**
+
+**MSS**
 
-      Use case ends.
+1. User requests to unlink a friend using a friend's `FRIEND_ID`, a game's `GAME_ID`.
+2. gitGud unlinks the friend from the game and informs user of successful unlinking of friend.
+
+   Use case ends.
+
+**Extensions**
 
+* 1a. gitGud detects that the `FRIEND_ID`/`GAME_ID` does not exist.
+    * 1a1. gitGud informs user that it has no friend/game was found with specified `FRIEND_ID`/`GAME_ID`.  
+    * 1a2. User makes another request to unlink a friend with a game, with a different `FRIEND_ID`/`GAME_ID` for an existing friend/game.  
+    * Use case continues from step 2.
 
-**Use case: UC04 - List games whose `GAME_ID` contains a keyword**
+* 1b. gitGud detects that the `FRIEND_ID` and `GAME_ID` link does not exist.
+    * 1b1. gitGud informs user that it has no friend and game link was found with specified `FRIEND_ID` and `GAME_ID`.  
+    * 1b2. User makes another request to unlink a friend with a game, with a different `FRIEND_ID` and/or `GAME_ID`.  
+    * Use case continues from step 2.
+
+**Use case: UC06 - Assign a skill value for a linked game to a friend**
```

Comment:
```suggestion
**Use case: UC06 - Assign a skill value for a friend to a linked game**
```

Reply:
Same as above,  I am trying to follow the UG's order, so I will edit the description below isntead.